zebra-chain/src/primitives/zcash_history.rs

```diff
@@ -106,12 +106,12 @@ impl<V: Version> Tree<V> {
     /// Will panic if `peaks` is empty.
     #[allow(clippy::unwrap_in_result)]
     pub fn new_from_cache(
-        network: Network,
+        network: &Network,
         network_upgrade: NetworkUpgrade,
         length: u32,
         peaks: &BTreeMap<u32, Entry>,
         extra: &BTreeMap<u32, Entry>,
-    ) -> Result<Self, io::Error> {
+    ) -> Result<Tree<V>, io::Error> {
```

Comment:
Nitpick: Unnecessary change / inconsistent.

```suggestion
    ) -> Result<Self, io::Error> {
```

Reply:
change applied